### ISSUE TITLE
Train and test Rasa models in Backoffice

### ISSUE THREAD

**Issue by turnerrainer** [buerokratt/Training-Module#386]
AS AN NLP Trainer
I WANT TO train and test new Rasa modules in Bürokratt Backoffice
SO THAT I could create new models and validate their quality before deploying them in production

### Acceptance Criteria

- [x] Set up [CronManager](https://github.com/buerokratt/CronManager) job for initiating Rasa training
- [ ] List of models with appropriate details stored in a Postgres database table called `llm_trainings`
- [ ] NLP models are stored outside of a Postgres database defined as a variable in a `settings` file

#### GUI

- [x] GUI based on https://admin.dev.buerokratt.ee/training/analytics/models

#### DSL

- [x] `GET` REST endpoint `/rasa/models` to list all available Rasa models
- [ ] `GET` REST endpoint `/rasa/model/{MODEL-ID}` to list a specific Rasa model
- [ ] `DELETE` REST endpoint `/rasa/model/{MODEL-ID}` to delete an existing Rasa model

**Comment by ExiRain** [comment]
[Pull request](https://github.com/buerokratt/Training-Module/pull/399) with changes.

**Comment by ValterAro** [comment]
Short summary: The page for training the model goes blank instantly when you click there.
How to reproduce: login and go to https://admin.dev.buerokratt.ee/training/train-new-model

Pic:


![Image](https://github.com/buerokratt/Training-Module/assets/122787093/1384d223-edb8-4da0-a805-a789725d3c90)

**Comment by ValterAro** [comment]
Short summary: The list of trained models is currently empty, but it should be populated perhaps. Vassili and Georg will take a look

Pic:


![Image](https://github.com/buerokratt/Training-Module/assets/122787093/3c219658-d0cd-4fee-9fda-f5ca1245cedb)

**Comment by ExiRain** [comment]
New [PR](https://github.com/buerokratt/Training-Module/pull/407): 
Fixed url for CRON, added safeguard to load page if no data to be shown.

Issues with current task:
@turnerrainer Guard still needs to be addressed, we temporally modified .guard in GET to allow requests to happen to ensure logic is working.
Long story training works,
@KlviG modified mock1 contents were modified in order to bot to train, because all temp data was made within training-module didn't allow bot to start.
@RayDNoper Did some changes to Cron-Manager to make it run, but CRON manager instance need to be fixed and path varibale feature added to it
Initialization of Cron-Manager need to be run in background, current solution is `nohup ./startup.sh &` or else w.e person runs the command and leaves the terminal would terminate the session.

Training module didn't had all db changes - FIXED
Long story short training logic is there, just separate deployed parts need some more attention.
One thing would be good to add it visual representation of whats currently happens with bot training, there many steps happen from initialization till the end and it takes some time(2 minutes+ depending on amount of data), so waiting for end response to pass it to GUI is not gonna do the trick.

**Comment by turnerrainer** [comment]
Needed redeployment.

**Comment by Kristjan259** [comment]
Gives 500 error code when going to the https://admin.dev.buerokratt.ee/training/analytics/models page. 



![Image](https://github.com/buerokratt/Training-Module/assets/5470201/17aa74e1-843a-4cd4-8a47-e884eb1f92ea)

**Comment by ExiRain** [comment]
Last bugfixes are under this [Pull Request](https://github.com/buerokratt/Training-Module/pull/407).
